### PR TITLE
fix(dev): New approach on get secrets

### PR DESCRIPTION
**Pull request by sindrig**
This closes island-is/infrastructure#933

### CODE REVIEW

File: infra/src/cli/render-secrets.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  services.forEach((svc) => {
+    Object.entries(svc).forEach(([serviceName, config]) => {
+      if (serviceName == service) {
+        Object.entries(config.secrets).forEach(([envName, ssmName]) => {
+          secretMap[ssmName] = envName
+        })
+      }
+    })
+  })
+  // There seem to be only 1 parameter that's labeled as dev?
```
Comment (sindrig) [inline infra/src/cli/render-secrets.ts:32]: @darrikonn I seem to remember you had something to do about this script at some point. Any idea why there is only 1 parameter labeled as dev now? Can we/should we simply drop that requirement?

Comment (darrikonn) [reply infra/src/cli/render-secrets.ts:32]: Yeah you're right. I wrote at least the first version of it.

There should be definitely more than 1 secret in dev labeled as dev.
Only 1 version had the label though. Might be that is what you're referring to?

Comment (sindrig) [reply infra/src/cli/render-secrets.ts:32]: I'm mixing tags and labels :facepalm: 
Reworking this....

Comment (sindrig) [reply infra/src/cli/render-secrets.ts]: Fixed now

File: infra/src/cli/render-secrets.ts [hunk 2]
```diff
@@ -2,79 +2,59 @@ import { generateYamlForEnv } from '../dsl/serialize-to-yaml'
...
-      throw new Error('params.Parameters aws undefined?')
-    }
-    return
-  }
 }
 
-const getAllDevParamsNames = async (): Promise<string[]> => {
-  const client = new ResourceGroupsTaggingAPI(API_INITIALIZATION_OPTIONS)
-  const names = []
+const getAllDevParams = async (): Promise<{ [name: string]: string }> => {
```
Comment (darrikonn) [inline infra/src/cli/render-secrets.ts:38]: Aren't there multiple environment variables that have the same name but for different projects (where the value might differ)?

Comment (sindrig) [reply infra/src/cli/render-secrets.ts:38]: This will return a map of `ssm-parameter-name: ssm-parameter-value`. SSM parameter names are unique.

Comment (darrikonn) [reply infra/src/cli/render-secrets.ts:38]: Yeah but then get-secrets will strip the environment variable name and load it using dotenv so they'll override each other if the final part of the aws param store variable matches.

Comment (sindrig) [reply infra/src/cli/render-secrets.ts:38]: Final part? What do you mean?

Let's say service A wants to map `/k8s/serviceA/SOME_VAR` to the environment variable `SOME_VAR` and you run `get-secrets serviceA`.

- serviceB wants to map `/k8s/serviceB/SOME_VAR` to the environment variable `NOT_SOME_VAR`. That won't override `SOME_VAR` as `NOT_SOME_VAR != SOME_VAR`
- serviceC wants to map `/k8s/serviceC/NOT_SOME_VAR` to the environment variable `SOME_VAR`. That WILL override `SOME_VAR` in `.env.secrets`, but that's exactly how the script behaved before so I'm not addressing that.

Comment (darrikonn) [reply infra/src/cli/render-secrets.ts:38]: Ahh sorry... I somehow assumed that you removed the service argument and you're just fetching every single environment variable, regardless of service.

**Review by darrikonn — APPROVED** [review]

**Pull request merged**